web-app/HeimScripts/heatmap/limmaUtils.R

```diff
@@ -33,3 +29,11 @@ getDesign <- function(measurements) {
 getSubset1Length <- function(measurements) {
   sum(grepl(pattern = SUBSET1REGEX, x = colnames(measurements))) # returns number of column names satisfying regexp.
 }
+
+# to check if row  contains at least one row that contains 3 non missing values
+rowContainsAtLeastThreeData <- function (row) {sum(!is.na(row)) > 2}
+
+# measurements should contains at least one valid row
+isValidLimmaMeasurements <- function (measurements) {
+   sum(apply(measurements, 1, rowContainsAtLeastThreeData)) > 1
```

Comment:
only if there are no rows with at least 3 measurements: in that case eBayes throws an error -- > it needs at least one row with remaining degrees of freedom >0 . if only 2 cells are not NA, then df=0...